sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/PushDownLeftSemiAntiJoin.scala

```diff
@@ -47,12 +47,17 @@ object PushDownLeftSemiAntiJoin extends Rule[LogicalPlan]
         p.copy(child = Join(gChild, rightOp, joinType, joinCond, hint))
       } else {
         val aliasMap = getAliasMap(p)
-        val newJoinCond = if (aliasMap.nonEmpty) {
-          Option(replaceAlias(joinCond.get, aliasMap))
+        // Do not push complex join condition
```

Comment:
Could you elaborate a little more explicitly about the condition instead of too general term, `complex`, @wangyum ?

Reply:
This PR simply bans all aliases and seems to aim to allow a plain column?

Reply:
> Could you please elaborate a little more explicitly about the condition instead of too general term, `complex`, 

complex means the children of this expression are non empty, another similar example:
https://github.com/apache/spark/blob/ad2bc7d82296527582dfa469aad33123afdf6736/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/PullOutGroupingExpressions.scala#L52-L59




> This PR simply bans all aliases and seems to aim to allow a plain column?
No. It will not ban this case:
```
    val originalQuery = testRelation
      .select($"a".as("new_a"))
      .join(testRelation1, joinType = LeftSemi, condition = Some($"new_a" === $"d"))
      .analyze
```

Reply:
Thanks.